pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -31,46 +77,46 @@ const (
 // match the init node, and either finishing if the status is reencrypt_finished, or applying the next secrets-encrypt
 // command.
 func (p *Planner) rotateEncryptionKeys(cp *rkev1.RKEControlPlane, clusterPlan *plan.Plan) error {
+	// ensure operating on latest control plane object
+	cp, err := p.rkeControlPlanes.Get(cp.Namespace, cp.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
 	if !shouldRotateEncryptionKeys(cp) {
 		return nil
 	}
 
 	if shouldRestartEncryptionKeyRotation(cp) {
 		return p.restartEncryptionKeyRotation(cp)
 	}
-	_, _, leader, err := p.findInitNode(cp, clusterPlan)
+
+	leader, err := p.encryptionKeyRotationElectLeader(cp, clusterPlan)
 	if err != nil {
 		return err
 	}
-	if leader == nil {
-		logrus.Error("Could not rotate encryption keys: unable to find init node")
-		return nil
-	}
+
+	logrus.Infof("rkecluster %s/%s current encryption key rotation phase command: [%s]", cp.Namespace, cp.Spec.ClusterName, cp.Status.RotateEncryptionKeysPhase)
```

Comment:
Should this be an `Info` level message?

Reply:
No, moved to debug. Now that I'm using `rke2.Ready` and checking for `"starting"` in the planner `ErrWaiting` return value, it is alot clearer.